### ISSUE TITLE
Revisit how to apply a yaml file or create a resource to this cluster

### ISSUE THREAD

**Issue by benoitf** [containers/podman-desktop#10161]
### UX Description

Today, it's possible to apply some yaml files (and with https://github.com/podman-desktop/podman-desktop/pull/10073) to create resources.

Discussing how it's working leads to some topics to iterate.

For example, 
- if you click on apply yaml file, you see a dialog prompt to select a file and the file is then applied
    - but here it would be nice to have a way to view the file before applying it (or to amed/edit it before applying)
    - in case of failure, you may be interested in staying on a page with the feedback (or the part of the file that have not been applied or something else)
- create resources, you have an editor full page and you can click on create
- in part of some mockups/discussions, it would be nice to have templates when creating resources (like create service and then you have the yaml file and you just replace some fragment)

![Image](https://github.com/user-attachments/assets/59a91a12-6ff7-47e7-8126-389a204fc3ed)

so, here it would be interesting to see if would make sense to have a unique way of doing almost the same operations in a Form page.

I click on a button, I go to a form page where I can import, edit, query, use templates, whatever, before sending the file to the kubernetes cluster to be applied

### Request type

Application mockups/designs

### Primary Contact

@benoitf

**Comment by ekidneyrh** [comment]
Just starting to have a look at this and I have some Qs.

Would we be looking for all those actions (import, edit, query, templates) under one button or have the APPLY import & edit and the CREATE use templates & edit? 

If so would the CREATE section be building on what @axel7083 has already done in the PR linked? 

I'm wondering if this ticket covers two elements or one? 

Some screenshots from Axels PR for reference:

![Image](https://github.com/user-attachments/assets/6ce697ca-41ab-472f-8b0e-b340c2423b9f)
![Image](https://github.com/user-attachments/assets/fea58022-e34e-4d5d-9a53-e2808ec30411)
![Image](https://github.com/user-attachments/assets/001176ff-afaf-4aa3-831b-f60e574b7ed8)

**Comment by deboer-tim** [comment]
IMHO Apply is likely to be more recognizable than Create, and still general enough. I see this as one form where you'd have several options to generate the yaml you want to apply (select a file to import, copy/paste, use a template, etc), and when you apply it shows a bit more detail about what was done (e.g. list of resources or links) instead of what the dialog shows (e.g. '2 of 3 applied').

**Comment by ekidneyrh** [comment]
I have a form mocked up with the options available to upload, use a template or copy/paste. I'm still brainstorming ways to have them not be separated and have the choice to build on from an upload or a template 

![Image](https://github.com/user-attachments/assets/62b552cc-5b0c-42d7-9351-bb03e523a02e)

**Comment by ekidneyrh** [comment]
another iteration that allows the user to choose either to upload a YAML or use a template, not both. If they choose to import either it will populate in the 'YAML editor' section where they can edit if they wish. 

![Image](https://github.com/user-attachments/assets/1615416a-1bf6-4c0b-aa34-a13b3b2f1d62)

I was wondering whether we intend on allowing the user to preview the templates before they import? And also how many templates will there be?

**Comment by deboer-tim** [comment]
> another iteration that allows the user to choose either to upload a YAML or use a template, not both. If they choose to import either it will populate in the 'YAML editor' section where they can edit if they wish.

I like this better. Even if we don't allow edit at first, it allows you to see the contents before applying. 

> I was wondering whether we intend on allowing the user to preview the templates before they import?

Same as above, I think this would allow them to preview and edit before applying.

> And also how many templates will there be?

We don't know at this point, would likely start with one per resource type.

**Comment by axel7083** [comment]
> If they choose to import either it will populate in the 'YAML editor' section where they can edit if they wish.

IMO The YAML editor is too small, you cannot edit much is such small section, when editing Kubernetes content :/ having it full page make sense for me

**Comment by ekidneyrh** [comment]
@axel7083 We could spread it out over two pages like so:

![Image](https://github.com/user-attachments/assets/1fc8cbf8-42c9-4d3f-98ed-8eeaf242bdc2)

If the last option is selected we can change the 'Import' button to 'next'.

![Image](https://github.com/user-attachments/assets/bf61bfbd-8b13-47da-8261-99506e735e54)

**Comment by axel7083** [comment]
I like the multi option page, `Upload YAML file`, `Use a Template`, `Create file from scratch`, but the editor page feel not like an editor. It feel small since it is a card, inside a details page 🤔 

Can't we make the editor using the full content of the details page? I was exploring this concept on the quadlet extension, here is something I POC

![Image](https://github.com/user-attachments/assets/d805273b-5fce-4df2-a058-ef303411ee24)

> The buttons are not clear, the button weird etc. but the idea of having the editor using all the content is nice (IMO)

**Comment by ekidneyrh** [comment]
@axel7083 Yah we can use the details page so it has more room. I would go with only having one tab for this case given it's for creating a YAML that will be used straight away. I was wondering if we have a save button - where would that save? On the podman application itself or on the users desktop? 

![Image](https://github.com/user-attachments/assets/be3bea38-f60e-4f19-b24a-21199ed885ce)

**Comment by axel7083** [comment]
@ekidneyrh in the cases of a full screen editor, I don't think we need to have a tab, or to allow the user to rename it.

The save options is probably not needed, we could probably just have the apply button as a start, I would expect that the save button allow me to save the yaml file to a directory => opening the file picker, and save it. (but this is like extra, not really needed)

Here is an example of the workflow of creating a Kubernetes (here a pod) resources in OpenLens

https://github.com/user-attachments/assets/d6429cd9-f205-48d1-855d-d4a9934dc6b4

In recent version of Lens, they seems to have change the way it is done, and honestly, I was a bit lost when I tried 

https://github.com/user-attachments/assets/92b2c13b-8945-409f-a5e7-b3e097d732f0

They use the `Save` icon to mean apply resources, which is a bit weird (IMO)

**Comment by deboer-tim** [comment]
The differences between our details pages and form pages aren't huge (some colours, buttons at the top, tabs, content border) aren't huge, but IMHO if we use them interchangeably or style the same there's little point in having both.

Compared to OpenLens or Lens the form space we'd have for an editor is a bit narrower but taller than either, which doesn't seem so bad to me. @axel7083 it's not clear to me if you're saying that's not enough, you don't like our forms (IMHO valid, but a separate issue), or we're getting off track here.

**Comment by ekidneyrh** [comment]
It doesnt look like much width is needed based on the examples sent by @axel7083 for creating a pod. 

Do we really need the full page of the details page for this particular use case? Or are we planning for future iterations where this component will be used elsewhere for other purposes? If it's for a general editor page, then a different issue may have t be logged so we can brainstorm all the possible use cases.

The reason I originally went with the form style page was bc it was part of the form flow, so it made more sense to have it similar.

**Comment by axel7083** [comment]
>[@axel7083](https://github.com/axel7083) it's not clear to me if you're saying that's not enough, you don't like our forms (IMHO valid, but a separate issue), or we're getting off track here.

@deboer-tim yeah, the issue was called `Revisit`, so I thought about exploring some other aspect, our form are okay, but they are not an editor, and we try to squeeze an editor logic inside our form, so this might be why it feel odd to me

**Comment by ekidneyrh** [comment]
Here's an updated mock-up with your suggestions @axel7083:

![Image](https://github.com/user-attachments/assets/9720781b-0218-4865-81f4-ba606c1d4eb0)

Is this the direction we want to go in?

**Comment by axel7083** [comment]
> Is this the direction we want to go in?

Yes I like it! But the buttons look a bit unclear!

**Comment by deboer-tim** [comment]
Discussed on UX call, we will implement this: https://github.com/podman-desktop/podman-desktop/issues/10161#issuecomment-2750759721 and then review how we are for space/usability. The button bar on the second page will be bottom-justified to make the yaml editor fill the screen and be as large as possible.

**Comment by nichjones1** [comment]
Closing as the implementation issue has been opened here: https://github.com/podman-desktop/podman-desktop/issues/12172
